### PR TITLE
Moved yesno filter test to its own file.

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: tests/template_tests/filter_tests/test_lower.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class FunctionTests(SimpleTestCase):
 
     def test_lower(self):
         self.assertEqual(lower('TEST'), 'test')
+        self.assertEqual(lower('test'), 'test')
```
Comment (atombrella) [inline tests/template_tests/filter_tests/test_lower.py]: I don't think this is necessary.

File: tests/template_tests/filter_tests/test_yesno.py [hunk 2]
```diff
@@ -1,6 +1,29 @@
 from django.template.defaultfilters import yesno
 from django.test import SimpleTestCase
 
+from ..utils import setup
+
+
+class YesNoTests(SimpleTestCase):
```
Comment (atombrella) [inline tests/template_tests/filter_tests/test_yesno.py]: There's already a single test for the `yesno`-filter in `django/tests/template_tests/syntax_tests/test_filter_syntax.py`. However, it's probably better to move them into a separate test-class. Please do this in a separate commit. Please see 7fce4dc5ff2d23bf43ee645f5bc0d6024ff1627f for an example.

File: tests/template_tests/filter_tests/test_upper.py [hunk 3]
```diff
@@ -26,6 +26,7 @@ class FunctionTests(SimpleTestCase):
 
     def test_upper(self):
         self.assertEqual(upper('Mixed case input'), 'MIXED CASE INPUT')
+        self.assertEqual(upper('MIXED CASE INPUT'), 'MIXED CASE INPUT')
```
Comment (atombrella) [inline tests/template_tests/filter_tests/test_upper.py]: I don't think this is necessary.

File: tests/template_tests/filter_tests/test_length.py [hunk 4]
```diff
@@ -43,6 +43,16 @@ def test_length07(self):
         output = self.engine.render_to_string('length07', {'None': None})
         self.assertEqual(output, '0')
 
+    @setup({'length08': '{{ dictionary|length }}'})
```
Comment (atombrella) [inline tests/template_tests/filter_tests/test_length.py]: Please choose more meaningful names for the template and test methods. The enumeration pattern is due to legacy reordering, as far as I understand.

**Comment by hramezani** [comment]
@atombrella, thanks for your comments. requested changes are done.

File: tests/template_tests/syntax_tests/test_filter_syntax.py [hunk 5]
```diff
@@ -109,135 +109,127 @@ def test_filter_syntax11(self):
...
-        output = self.engine.render_to_string('filter-syntax13', {"var": SomeClass()})
+        output = self.engine.render_to_string('filter-syntax12', {"var": SomeClass()})
         if self.engine.string_if_invalid:
             self.assertEqual(output, "1INVALID2")
         else:
             self.assertEqual(output, "12")
 
-    @setup({'filter-syntax14': r'1{{ var.method4 }}2'})
-    def test_filter_syntax14(self):
+    @setup({'filter-syntax13': r'1{{ var.method4 }}2'})
```
Comment (atombrella) [inline tests/template_tests/syntax_tests/test_filter_syntax.py]: I don't think you should re-number the existing tests.

**Comment by hramezani** [comment]
@atombrella thanks, re-numberd tests reverted.

**Comment by timgraham** [comment]
I don't see much need to duplicate every function test as a template test. What's your thinking about this?

**Comment by hramezani** [comment]
@timgraham I just want to double check it. I can remove extra tests. please tell me extra tests name, I will remove them.

**Comment by hramezani** [comment]
@timgraham , I removed two function tests in `tests/template_tests/filter_tests/test_length.py`. if there are extra tests please tell me to remove them.

**Comment by timgraham** [comment]
What I mean is that test_add_date_with_timedelta duplicates test_add07, test_add_tuple duplicates test_add06, and similar.

**Comment by hramezani** [comment]
@timgraham, Sorry for being late. I removed above-mentioned duplicate tests:

- test_add_date_with_timedelta 
- test_add_date_with_timedelta

**Comment by timgraham** [comment]
I don't see the value in the extra tests. Generally, we don't need to duplicate function and filter tests. One or the other is usually sufficient. Some are duplicated now but we don't need to delete the duplicates or add more duplicates. I'll merge the commit that moves the one test.
